pkg/gadgettracermanager/bpf-maps.h

```diff
@@ -0,0 +1,24 @@
+/* SPDX-License-Identifier: (GPL-2.0 WITH Linux-syscall-note) OR Apache-2.0 */
+
+#ifndef GADGET_TRACER_MANAGER_MAPS_H
+#define GADGET_TRACER_MANAGER_MAPS_H
+
+#include "common.h"
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__type(key, __u64);
+	__type(value, struct container);
+	__uint(max_entries, MAX_CONTAINER_PER_NODE);
+} containers SEC(".maps");
+
+#ifdef WITH_FILTER
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__type(key, __u64);
+	__type(value, __u32);
+	__uint(max_entries, MAX_CONTAINER_PER_NODE);
+} filter SEC(".maps");
+#endif
```

Comment:
I'm a bit concerned about having multiple versions of the compiled code around. I think it increases the complexity and could be confusing in some cases. Also if we have more than one "optional" map, we'd have a lot of different combinations.

I tried to use a different approach to avoid having this conditional compilation. The idea is to have a global variable in the eBPF program to enclose the access to the map, then set this to false and replace the map by a dummy one to avoid creating it when loading the program. An example of this approach is available [here](https://github.com/kinvolk/ebpf/commits/mauricio/disable-maps-idea).  Do you think it's worth to try it out or do you prefer to keep using the conditional compilation approach?

Reply:
I think the global variable + dummy map approach increases the complexity too in the Go code because it needs to be generated. And are you sure that the verifier can remove the dead code around the map usage?

Reply:
Ah got it about the generated code, yeah, without that it's a mess. I agree to keep using the conditional compilation.

Reply:
About the verifier, yes, it does remove the dead code.